OrbitCore/CaptureData.cpp

```diff
@@ -12,3 +16,28 @@ orbit_client_protos::LinuxAddressInfo* CaptureData::GetAddressInfo(
   }
   return &address_info_it->second;
 }
+
+const FunctionStats& CaptureData::GetOrInsertFunctionStats(
+    uint64_t function_address) {
+  return functions_stats_[function_address];
+}
+
+void CaptureData::UpdateFunctionStats(
+    orbit_client_protos::FunctionInfo* func,
+    const orbit_client_protos::TimerInfo& timer_info) {
+  const uint64_t function_address = func->address();
+  FunctionStats& stats = functions_stats_[function_address];
```

Comment:
It is always better to create a temporary object - compute it and as final stem insert to the map. The reason is that if something goes wrong we do not end up with half baked instance in the map.

Reply:
actually - this is not relevant in this case since it looks like this is updating existing record.

Reply:
Right, its updating the existing value or if this is the first update, create empty stats and then update it.